File: 1 Python/Codeforces-1806A-WalkingMaster.py
nc = int(input())

for _ in range(nc):
    a,b,c,d = map(int, input().rstrip().split())
    vert = d-b
    horiz = c-a

    if horiz>vert or vert < 0:
        ans = -1
    else:
        ans = 2*vert - horiz

    print(ans)

File: 2 CPP/Codeforces-1806A-WalkingMaster.cpp
#include <bits/stdc++.h>
using namespace std;

void solve(){
    long a,b,c,d;
    cin >> a >> b >> c >> d;
    long vert = d-b, horiz = c-a;
    long ans = (horiz>vert || vert<0) ? -1:2*vert-horiz;
    cout << ans << "\n";
}

int main(){
    int tc;
    cin >> tc;
    while(tc--){
        solve();
    }
    return 0;
}
